src/Map/src/MapFactory.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Map;
+
+use Symfony\UX\Map\Provider\ProviderInterface;
+
+/**
+ * Creates a map based on the configuration, and registers it in the map registry.
+ *
+ * @author Hugo Alliaume <hugo@alliau.me>
+ */
+final readonly class MapFactory implements MapFactoryInterface
```

Comment:
+ @internal

Reply:
Hum, yes and no, maybe you thought about really making the `MapFactory` internal and use something else to create the map in user-land?
<img width="817" alt="image" src="https://github.com/symfony/ux/assets/2103975/55e090ff-2734-4254-9446-db56ebf3ba99">

Reply:
Fixed, now we can simply call `new Map()`